src/testUtils/mocks/dataSources.ts

```diff
@@ -2501,3 +2506,52 @@ export const createMockScheduleSpec = (scheduleSpec?: {
     !scheduleSpec
   ) as ScheduleSpec;
 };
+
+/**
+ * @hidden
+ * NOTE: `isEmpty` will be set to true if no value is passed
+ */
+export const createMockScalar = (scalar?: string): Scalar =>
+  createMockStringCodec(scalar) as Scalar;
+
+export const createMockRistrettoPoint = (ristrettoPoint?: string): RistrettoPoint =>
+  createMockStringCodec(ristrettoPoint) as RistrettoPoint;
+
+export const createMockSignature = (signature?: string): Signature =>
+  createMockStringCodec(signature) as Signature;
+
+export const createMockZkProofData = (
+  challengeResponses: string[],
+  subtractExpressionsRes: string,
+  blindedScopeDidHash: string
+): ZkProofData => {
+  return createMockCodec(
+    {
+      challenge_responses: challengeResponses.map(cr => createMockScalar(cr)),
+      subtract_expressions_res: createMockRistrettoPoint(subtractExpressionsRes),
+      blinded_scope_did_hash: createMockRistrettoPoint(blindedScopeDidHash),
+    },
+    false
+  ) as ZkProofData;
+};
+
+export const createMockScopeClaimProof = (
```

Comment:
See how the other struct-creating functions are implemented (`createMockDocument` for example). You need to allow passing nothing to allow simulating empty values

Reply:
I saw them but in this case, is there any use case for empty values? I couldn't think any of them

Reply:
At least the interface should be similar, receiving an object instead of a list of values